zk-token-sdk/src/errors.rs

```diff
@@ -1,55 +1,50 @@
 //! Errors related to proving and verifying proofs.
 use {
-    crate::{range_proof::errors::RangeProofError, sigma_proofs::errors::*},
+    crate::{
+        encryption::elgamal::ElGamalError,
+        range_proof::errors::{RangeProofGenerationError, RangeProofVerificationError},
+        sigma_proofs::errors::*,
+    },
     thiserror::Error,
 };
 
 #[derive(Error, Clone, Debug, Eq, PartialEq)]
-pub enum ProofError {
-    #[error("invalid transfer amount range")]
-    TransferAmount,
+pub enum ProofGenerationError {
     #[error("proof generation failed")]
```

Comment:
nit: this error can be more precise now, maybe for simplicity: 
```suggestion
    #[error("not enough funds in account")]
```